iroh-relay/src/ping_tracker.rs

```diff
@@ -0,0 +1,64 @@
+use std::time::{Duration, Instant};
+
+use tracing::debug;
+
+/// Tracks pings on a single relay connection.
+///
+/// Only the last ping needs is useful, any previously sent ping is forgotten and ignored.
+#[derive(Debug)]
+pub struct PingTracker {
+    inner: Option<PingInner>,
+    default_timeout: Duration,
+}
+
+#[derive(Debug)]
+struct PingInner {
+    data: [u8; 8],
+    deadline: Instant,
+}
+
+impl PingTracker {
+    /// Creates a new ping tracker, setting the ping timeout for pings.
+    pub fn new(default_timeout: Duration) -> Self {
```

Comment:
Why make the timeout configurable if both users use the same timeout?

Also both users define their own constant for the timeout right now.  I'd be tempted to put the constant in this module and hardcode it again.  But I don't feel very strongly about this.

Reply:
I am not against that, I just wasn't sure we'd always want to have the same numbers 🤷